src/PotentialCalculation/PotentialCalculationSetup/BoundaryConditions/BoundaryConditionsCylindrical.jl

```diff
@@ -30,13 +30,47 @@ function apply_boundary_conditions_on_r_axis!(  rbpot::AbstractArray{T, 4}, rbi:
     nothing
 end
 
+function apply_boundary_conditions_at_r0!(
+    rbpot::AbstractArray{T, 4}, 
+    rbi::Int,
+    nnz::Int,
+    gw::AbstractArray{T},
+    update_even_points::Val{even_points}
+    ) where {T, even_points}
+    if even_points
```

Comment:
Is julia smart enough to compile this away as this is known from the arguments passed to the method?

Reply:
Yes, as the boolean is passed as a type via `Val`.